### PR TITLE
Convert ddim

### PR DESCRIPTION
**Pull request by freedomtan**
1. colab [notebook](https://colab.research.google.com/drive/1120anEh9Js9tPEPod-M3-w-x1UFcUZdS?usp=sharing) 
2. diff f98790b
  - convert keras/tf.keras to keras_core
  - replace most of tf ops except tf.image, tf.data, and tf.GrapdientTap
  - fix ops.concat out of range issue

### CODE REVIEW

File: keras_core/layers/merging/concatenate.py [hunk 1]
```diff
@@ -69,7 +69,8 @@ def build(self, input_shape):
                 if axis != concat_axis and axis_value == 1:
                     del reduced_inputs_shapes[i][axis]
 
-            del reduced_inputs_shapes[i][self.axis]
+            if len(reduced_inputs_shapes[i]) > self.axis:
```
Comment (fchollet) [inline keras_core/layers/merging/concatenate.py:72]: Why did this come up?

**Review by fchollet — COMMENTED** [review]
Thanks for the PR! Looking good!

File: keras_core/layers/merging/concatenate.py (hunk 1 above)
Comment (freedomtan) [reply keras_core/layers/merging/concatenate.py:72]: In the example, in Inception V3, some axes were squeezed out, and thus, there were error messages showing out-of-range while accessing the `reduced_inputs_shapes[i][self.axis]`. Alternatives I could think of:
1. Don’t squeeze
2. When squeezing, also decrement `self.axis`

**Review by fchollet — APPROVED** [review]
LGTM
